flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -206,7 +206,9 @@ public RexNode visitTypeLiteral(TypeLiteralExpression typeLiteral) {
 
 	@Override
 	public RexNode visit(Expression other) {
-		if (other instanceof ResolvedAggInputReference) {
+		if (other instanceof UnresolvedFieldReferenceExpression) {
```

Comment:
I need some clarification here. AFAIK, the only "unresolved" expressions only exist during the phase we constructing table API in original Flink, right? The unresolved expression never get a chance to translate to Calcite's RexNode. It should be resolved before that or an exception will throw. 

Are we changing this now? If we want to change this, it will be a little ambiguous with who are responsible for resolving all these expressions. Not sure "ExpressionVisitor" is the one, since it can see unresolved & resolved expressions in the same time.

Reply:
Yes I agree `UnresolvedFieldReference` cannot be translated into RexNode and also that it should never end up in the Planner. This is also the main purpose of the `UnresolvedFieldReference` to make this distinction clear.

`ExpressionVisitor` is just a way to traverse the operation tree. I am currently working on moving all the resolution of expressions into a single place in the API module, as part of FLINK-11884. I agree we should move this resolution here up to the API module as well once it's in place.

Reply:
Thanks for the clarification, +1 to "move this resolution here up to the API module"